docs/spec/v1/gitrepositories.md

```diff
@@ -319,6 +319,12 @@ Valid examples are: `refs/heads/main`, `refs/tags/v0.1.0`, `refs/pull/420/head`,
 This field takes precedence over [`.branch`](#branch-example),
 [`.tag`](#tag-example), and [`.semver`](#semver-example).
 
+**Note:** Azure Devops and AWS CodeCommit don't support fetching the HEAD of
+a PR. While Azure Devops lets you fetch the merge commit that will be created
+after merging a PR (using `refs/pull/<id>/merge`), this field can only be used
+to fetch references that exist in the current state of the Git repository and
+not references that will be created in the future.
```

Comment:
What does this effectively mean?

Reply:
azure devops support fetching the ref `refs/pull/<id>/merge`. this ref points to the merge commit that will be created after you merge the commit. i'm unsure about whether `created in the future` is a technically correct description of what's going on but i think its practically correct to think about it like that. essentially, we support `refs/pull/1/head`, since that points to the HEAD of the PR, which is a commit that exists at the present moment, but the former points to a commit that essentially does not exist at the present moment

Reply:
Small suggestions then to address styling nits:

```suggestion
**Note:** Azure DevOps and AWS CodeCommit do not support fetching the HEAD of
a pull request. While Azure DevOps allows you to fetch the merge commit that will be
created after merging a PR (using `refs/pull/<id>/merge`), this field can only be used
to fetch references that exist in the current state of the Git repository and
not references that will be created in the future.
```